sw/device/lib/base/hardened.h

```diff
@@ -32,4 +36,132 @@ typedef enum hardened_bool {
   kHardenedBoolFalse = 0x1d4u,
 } hardened_bool_t;
 
+/**
+ * Launders the 32-bit value `val`.
+ *
+ * `launder32()` returns a value identical to the input, while introducing an
+ * optimization barrier that prevents the compiler from learning new information
+ * about the original value, based on operations on the laundered value. This
+ * does not work the other way around: some information that the compiler has
+ * already learned about the value may make it through; this last caveat is
+ * explained below.
+ *
```

Comment:
Can you introduce the motivation here? I'm thinking something like this might help set the scene a bit:

```c
It is desirable in some circumstances to introduce redundant operations
into the instruction stream in order to make it less likely that hardware
faults will result in critical operations being bypassed. The `launder32`
function makes it possible to insert such duplicate operations without the
compiler necessarily removing them as dead code.
```